lib/publiccloud/provider.pm

```diff
@@ -41,10 +41,34 @@ sub init {
     my ($self) = @_;
     my $file = lc get_var('PUBLIC_CLOUD_PROVIDER');
     assert_script_run('mkdir -p ' . TERRAFORM_DIR);
-    assert_script_run('curl ' . data_url('publiccloud/terraform/' . $file . '.tf') . ' -o ' . TERRAFORM_DIR . '/plan.tf');
+    if (check_var('SLES4SAP_MODE', 'publiccloud')) {
+        my $git_repo   = get_var('ha-sap-terraform-deployments');
+        my $cloud_name = $self->get_sap_cloud_name;
+        assert_script_run('cd ' . TERRAFORM_DIR);
+        assert_script_run('git clone --depth 1 --branch ' . get_var('HA_SAP_GIT_TAG', 'master') . ' ' . get_required_var('HA_SAP_GIT_REPO') . ' .');
+        assert_script_run('cp pillar_examples/automatic/{cluster.sls,hana.sls,top.sls} salt/hana_node/files/pillar/');
+        assert_script_run('cd');    # We need to ensure to be in the home directory
+        assert_script_run('curl ' . data_url("publiccloud/terraform/sap/$file.tfvars") . ' -o ' . TERRAFORM_DIR . "/$cloud_name/terraform.tfvars");
+        $self->create_ssh_key(ssh_private_key_file => TERRAFORM_DIR . '/salt/hana_node/files/sshkeys/cluster.id_rsa');
+    }
+    else {
+        assert_script_run('curl ' . data_url("publiccloud/terraform/$file.tf") . ' -o ' . TERRAFORM_DIR . '/plan.tf');
+    }
     $self->create_ssh_key();
 }
 
+=head2 get_sap_cloud_name
+
+Does the conversion between C<PUBLIC_CLOUD_PROVIDER> and Terraform providers name.
+
+=cut
+sub get_sap_cloud_name {
+    # Check https://github.com/SUSE/ha-sap-terraform-deployments/issues/177 for more information
+    my $cloud_provider = lc get_var('PUBLIC_CLOUD_PROVIDER');
+    return 'aws' if $cloud_provider eq 'ec2';
```

Comment:
@cfconrad  any problem from our side to rename providers ?

Reply:
your sure @ldevulder  no problem with your workaround but after reading conversation in  https://github.com/SUSE/ha-sap-terraform-deployments/issues/177 I wonder why we have it different

Reply:
Historical reasons I would say :D More seriously for AWS for example I think it's because EC2 is the name for the VM part and AWS is more generic. But yes as Terraform can handle the whole AWS stack the AWS name is more logical.

Reply:
Don't see a problem with renaming it. But also don't see much benefit of it, as terraform is only one possibility to get a running VM. Previously we used ipa and I guess it will not be our last choice :)
I guess the name comes initial from the name of the images we get, these have GCE and EC2 in there names.

The only thing which I would change is the name of the method, so it really makes sure this is a translation from openQA name to terraform provider extension.

Reply:
> Or what was the reason to not use the openQA names for your terraform folders?
The reason is because we used the provider plugin names of Terraform, as explained here: https://github.com/SUSE/ha-sap-terraform-deployments/issues/177.

So I think the easiest way will be to keep this "workaround", and yes I can rename the function if you want something more general. Any suggestion for a new name that could fit for you?

Reply:
> The only thing which I would change is the name of the method, so it really makes sure this is a translation from openQA name to terraform provider extension.

`conv_openqa_tf_name` could be better?